src/ILVerification/src/TypeVerifier.cs

```diff
@@ -98,36 +99,59 @@ public void VerifyInterfaces()
                         MethodDesc resolvedMethod = type.ResolveInterfaceMethodTarget(method);
                         if (resolvedMethod is null)
                         {
-                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType), Format(method));
+                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType, implementedInterface.InterfaceImplementation), Format(method));
                         }
                     }
                 }
             }
         }
 
-        // Format helpers for future use
-        // The idea is to improve formatting string with tokens value i.e.: 
-        // [Assembly]Class(0x1234) vs current [Assembly]Class
-        // in case somebody will pass '--tokens' switch to ILVerify
         private string Format(EcmaType type)
         {
-            return type.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return string.Format("{0}([{1}]0x{2:X8})", type, _module, _module.MetadataReader.GetToken(_typeDefinitionHandle));
```

Comment:
This should not be using `_module`. This should get module from the type - same way as `Format(MethodDesc methodDesc)` gets the module from the method.

Reply:
Done, I also renamed some vars to be more clear and passed "Module" also on InterfaceImpl print token to have similar style(no reference to local field)

Reply:
I could try to understand if it's possible get module from InterfaceImplementation to avoid parameter.

Reply:
I do not think it is possible.